### PR TITLE
Subdevice status messages

### PR DESCRIPTION
**Pull request by nomis52**
Blocked behind #107

### CODE REVIEW

**Comment by nomis52** [comment]
This one is ready now as well.

File: firmware/src/dimmer_model.c [hunk 1]
```diff
@@ -203,8 +230,99 @@ bool ResetToBlockAddress(uint16_t start_address) {
...
+  ptr = PushUInt16(ptr, message->data_value2);
+  return ptr;
+}
+
+void QueueStatusMessage(DimmerSubDevice *device,
+                        RDMStatusType status_type,
+                        RDMStatusMessageId status_id,
+                        uint16_t data_value1,
+                        uint16_t data_value2) {
+  if (device->sd_report_threshold == STATUS_NONE &&
```
Comment (peternewman) [inline firmware/src/dimmer_model.c]: Shouldn't this be or?

File: firmware/src/dimmer_model.c [hunk 2]
```diff
@@ -203,8 +230,99 @@ bool ResetToBlockAddress(uint16_t start_address) {
...
+  uint8_t threshold = param_data[0];
+
+  if (threshold > STATUS_ERROR) {
+    return RDMResponder_BuildNack(header, NR_DATA_OUT_OF_RANGE);
+  }
+
+  uint8_t *ptr = g_rdm_buffer + sizeof(RDMHeader);
+
+  if (threshold == STATUS_GET_LAST_MESSAGE) {
+    // Return the last ones
```
Comment (peternewman) [inline firmware/src/dimmer_model.c]: I thought you only return one?

File: firmware/src/dimmer_model.c [hunk 3]
```diff
@@ -203,8 +230,99 @@ bool ResetToBlockAddress(uint16_t start_address) {
...
+    unsigned int i = 0u;
+    for (; i < NUMBER_OF_SUB_DEVICES; i++) {
+      DimmerSubDevice *subdevice = &g_subdevices[i];
+      StatusMessage *msg = &subdevice->status_message;
+
+      if (!msg->is_active) {
+        continue;
+      }
+
+      if (threshold < (msg->status_type & 0x0f)) {
```
Comment (peternewman) [inline firmware/src/dimmer_model.c]: Make 0x0f a constant.

**Comment by peternewman** [comment]
Generally LGTM

File: firmware/src/dimmer_model.c (hunk 1 above)
Comment (nomis52) [inline firmware/src/dimmer_model.c]: Good catch!

File: firmware/src/dimmer_model.c (hunk 2 above)
Comment (nomis52) [inline firmware/src/dimmer_model.c]: You can pack many status messages into a response. Maybe you're thinking of queued messages?

File: firmware/src/dimmer_model.c (hunk 3 above)
Comment (nomis52) [inline firmware/src/dimmer_model.c]: Done

File: firmware/src/dimmer_model.c (hunk 2 above)
Comment (peternewman) [inline firmware/src/dimmer_model.c]: I think I was mostly confused by the constant STATUS_GET_LAST_MESSAGE.

**Comment by peternewman** [comment]
LGTM

**Pull request merged**